compiler/ballerina-tools-api/src/main/java/io/ballerina/tools/diagnostics/DefaultDiagnostic.java

```diff
@@ -45,6 +47,14 @@ class DefaultDiagnostic extends Diagnostic {
         this.message = MessageFormat.format(diagnosticInfo.messageFormat(), args);
     }
 
+    public DefaultDiagnostic(DiagnosticInfo diagnosticInfo,
```

Comment:
Shall we improve the PackageDiagnostic class to support null locations without touching this class?

Reply:
Removed changes were done to the `DefaultDiagnostic` class and improved `PackageDiagnostic` class.